plugins/woocommerce/src/Admin/RemoteInboxNotifications/RemoteInboxNotificationsEngine.php

```diff
@@ -233,4 +245,79 @@ public static function get_note_from_db( $note_from_db ) {
 
 		return $note_from_db;
 	}
+
+	/**
+	 * Add the debug tools to the WooCommerce debug tools (WooCommerce > Status > Tools).
+	 *
+	 * @param array $tools a list of tools.
+	 *
+	 * @return mixed
+	 */
+	public static function add_debug_tools( $tools ) {
+		$tools['refres_remote_inbox_notifications'] = array(
+			'name'     => __( 'Refresh Remote Inbox Notifications', 'woocommerce' ),
+			'button'   => __( 'Refresh', 'woocommerce' ),
+			'desc'     => __( 'This will refresh the remote inbox notifications', 'woocommerce' ),
+			'callback' => function () {
+				RemoteInboxNotificationsDataSourcePoller::get_instance()->read_specs_from_data_sources();
+				RemoteInboxNotificationsEngine::run();
+
+				return __( 'Remote inbox notifications have been refreshed', 'woocommerce' );
+			},
+		);
+
+		$tools['delete_remote_notification'] = array(
+			'name'     => __( 'Delete Remote Notification', 'woocommerce' ),
+			'button'   => __( 'Delete', 'woocommerce' ),
+			'desc'     => __( 'This will delete a remote notification by slug', 'woocommerce' ),
+			'selector' => array(
+				'description'   => __( 'Select a notification to delete:', 'woocommerce' ),
+				'class'         => 'wc-product-search',
+				'search_action' => 'woocommerce_json_remote_inbox_notifications_search',
+				'name'          => 'delete_remote_notification_note_id',
+				'placeholder'   => esc_attr__( 'Search for a notification&hellip;', 'woocommerce' ),
+			),
+			'callback' => function () {
+				check_ajax_referer( 'debug_action', '_wpnonce' );
+
+				if ( ! isset( $_GET['delete_remote_notification_note_id'] ) ) {
+					return __( 'No remote inbox notification selected', 'woocommerce' );
+				}
+				$note_id = wc_clean( sanitize_text_field( wp_unslash( $_GET['delete_remote_notification_note_id'] ) ) );
+				$note = Notes::get_note( $note_id );
+				$note->delete( true );
```

Comment:
Maybe we'll need to check the note's existence here

Reply:
Ah..that's right! It throws exception. I'll update it!

Reply:
@ilyasfoo  Fixed in [Do not attempt to delete a note if it is not found](https://github.com/woocommerce/woocommerce/pull/48961/commits/4457d67bdf6c49609f96d10ac327f0697141d152)